x-pack/plugins/observability/public/pages/alerts/containers/alerts_table_t_grid/alerts_table_t_grid.tsx

```diff
@@ -336,11 +335,11 @@ export function AlertsTableTGrid(props: AlertsTableTGridProps) {
           pick(c, ['columnHeaderType', 'displayAsText', 'id', 'initialWidth', 'linkField'])
         ),
       });
-      if (newState !== localStorage.getItem(ALERT_TABLE_STATE_STORAGE_KEY)) {
-        localStorage.setItem(ALERT_TABLE_STATE_STORAGE_KEY, newState);
+      if (newState !== localStorage.getItem(stateStorageKey)) {
+        localStorage.setItem(stateStorageKey, newState);
```

Comment:
Could you help me understand why we keep altering the localStorage value for the alert table? Like this aren't we changing the o11y alerts table settings to match the other TGrid settings which we don't want?

Reply:
@fkanout Sure I'll try to explain. The only change I did here was to:
- not use the common `ALERT_TABLE_STATE_STORAGE_KEY` that was used in both `Overview` and `Alerts` page and was causing this bug https://github.com/elastic/kibana/issues/126765, 
- but use the `stateStorageKey` which is a **new prop** I introduced to the `AlertsTableTGrid` component

So now the two consumers [alerts page](https://github.com/elastic/kibana/pull/130538/files#diff-ab4cbe22bf9a1c9bbaf3a3cb32b5aa5b0c8b8e33c34b88ed3c99dbde6fedadb9R333) and [old overview page](https://github.com/elastic/kibana/pull/130538/files#diff-49584f9f351ad4371a9ac21128445da30c5d42fde875b85705069e5360534230R198) pass the prop to the AlertsTableTGrid component. Nothing else changed in the previous logic, just the parameterization of the localStorage key. 

Regarding the existing localStorage functionality (which didn't change in this PR) it works in a way that when the `tGridState` is changed, it will update the localStorage to the new user settings, if they are of course different from what is currently saved in that localStorage key. `tGridState` will be changed when any of the following is changed [onStateChange](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/alerts/containers/alerts_table_t_grid/alerts_table_t_grid.tsx#L375) (columns, sort, selectedEventIds). You can play around a bit with the UI and change table columns or sort for example and see what is saved in localStorage.